subs/pantry/src/Pantry/Storage.hs

```diff
@@ -437,30 +459,119 @@ loadHackageTarballInfo name version = do
   where
     go (Entity _ ht) = (hackageTarballSha ht, hackageTarballSize ht)
 
+getKey :: Either (Entity record) (Key record) -> Key record
+getKey dat = case dat of
```

Comment:
nit: not a very descriptive name.
Then again, `getKeyFromEntityOrItself` is not really better.

Reply:
I don't have strong opinion on either of the names. While the current name isn't great, I feel it's fine. The function implementation is small enough and it isn't exported to other modules. Although, if you any other nice name for the function and describes more concisely, I think we can rename it.

Reply:
I'll just replace the function with `either entityKey id`, which is pretty common in persistent code.